SRC/dlassq.f90

```diff
@@ -189,12 +200,14 @@ subroutine DLASSQ( n, x, incx, scl, sumsq )
    if( sumsq > zero ) then
       ax = scl*sqrt( sumsq )
       if (ax > tbig) then
-         abig = abig + (ax*sbig)**2
+!        We assume scl >= sqrt( TINY*EPS ) / sbig
+         abig = abig + (scl*sbig)**2 * sumsq
```

Comment:
What happens if `scl=1` and `sumsq=inf` (assuming inf is a valid input for sumsq)? In the previous version, ax would be inf and so would be abig. In the new version, `sbig**2` underflows to 0, and times inf gives abig = NaN. I think this is unexpected and undesired.

I've made these changes in our Go port and this is just one of many test cases that started to fail and that is easy to reason about. I haven't looked into the rest, I wanted to share this quickly before the PR is merged.

Reply:
> What happens if `scl=1` and `sumsq=inf` (assuming inf is a valid input for sumsq)? In the previous version, ax would be inf and so would be abig. In the new version, `sbig**2` underflows to 0, and times inf gives abig = NaN. I think this is unexpected and undesired.

`scl=1` and `sumsq=inf` would be an invalid input for the proposed change since, in this case, `1 < sqrt( TINY*EPS ) / sbig`. If we want to deal if this case, we should add another if-tense inside the scope of `if (ax > tbig) then`.

Do you think this is an important use case for LASSQ?

Reply:
> Do you think this is an important use case for LASSQ?

Not at all. I've re-read your comments above about why this change is what it is and now it all clicked. IIUC, the tradeoff is a slightly limited input range for better accuracy. I will adjust our tests accordingly.

Reply:
Just for fun. If we want to support all input ranges, the code would be something like:

```f90
!
!  Put the existing sum of squares into one of the accumulators
!
   if( sumsq > zero ) then
      ax = scl*sqrt( sumsq )
      if (ax > tbig) then
         if ( scl >= sqrt( TINY*EPS ) / sbig ) then
            abig = abig + (scl*sbig)**2 * sumsq
         else if ( scl < tsml ) then
            abig = abig + (scl*ssml)**2 * (sbig/ssml)**2 * sumsq
         else
            abig = abig + (scl**2 * sbig) * (sbig * sumsq)
         end if
      else if (ax < tsml) then
         if (notbig) then
           if ( scl <= sqrt( HUGE ) / ssml ) then
             ...
```

Maybe a little smaller if `sqrt( TINY*EPS ) / sbig <= sqrt( HUGE ) / ssml`. I didn't check. :)

Reply:
The upside of this would be that it would avoid the need to document all the constants and thresholds in the documentation for DLASSQ but I'm not in a position to decide whether this should be done or not. Feel free to re-approve and merge.